interop-testing/src/main/java/io/grpc/testing/integration/StressTestClient.java

```diff
@@ -0,0 +1,581 @@
+    threadpool = MoreExecutors.listeningDecorator(newFixedThreadPool(numThreads));
+    int server_idx = -1;
+    for (InetSocketAddress address : addresses) {
+      server_idx++;
+      for (int i = 0; i < channelsPerServer; i++) {
+        ManagedChannel channel = createChannel(address);
+        channels.add(channel);
+        for (int j = 0; j < stubsPerChannel; j++) {
+          String gaugeName =
+              String.format("/stress_test/server_%d/channel_%d/stub_%d/qps", server_idx, i, j);
+          Worker worker =
+              new Worker(channel, testCaseWeightPairs, durationSecs, gaugeName);
+
+          workerFutures.add(threadpool.submit(worker));
+        }
+      }
+    }
+  }
+
+  @VisibleForTesting
+  void blockUntilStressTestComplete() throws Exception {
+    Preconditions.checkState(!shutdown, "client was shutdown.");
+
+    ListenableFuture<?> f = Futures.allAsList(workerFutures);
+    if (durationSecs == -1) {
+      // '-1' indicates that the stress test runs until terminated by the user.
+      f.get();
+    } else {
+      f.get(durationSecs + WORKER_GRACE_PERIOD_SECS, SECONDS);
+    }
+  }
+
+  @VisibleForTesting
+  void shutdown() {
+    if (shutdown) {
+      return;
+    }
+    shutdown = true;
+
+    for (ManagedChannel ch : channels) {
+      try {
+        ch.shutdownNow();
+        ch.awaitTermination(1, SECONDS);
+      } catch (Throwable t) {
+        log.log(Level.WARNING, "Error shutting down channel!", t);
+      }
+    }
+
+    try {
+      metricsServer.shutdownNow();
+    } catch (Throwable t) {
+      log.log(Level.WARNING, "Error shutting down metrics service!", t);
+    }
+
+    try {
+      if (threadpool != null) {
+        threadpool.shutdownNow();
+      }
+    } catch (Throwable t) {
+      log.log(Level.WARNING, "Error shutting down threadpool.", t);
+    }
+  }
+
+  private static List<InetSocketAddress> parseServerAddresses(String addressesStr) {
+    List<InetSocketAddress> addresses = new ArrayList<InetSocketAddress>();
+
+    for (List<String> namePort : parseCommaSeparatedTuples(addressesStr)) {
+      String name = namePort.get(0);
+      int port = Integer.valueOf(namePort.get(1));
+      addresses.add(new InetSocketAddress(name, port));
+    }
+
+    return addresses;
+  }
+
+  private static List<TestCaseWeightPair> parseTestCases(String testCasesStr) {
+    List<TestCaseWeightPair> testCaseWeightPairs = new ArrayList<TestCaseWeightPair>();
+
+    for (List<String> nameWeight : parseCommaSeparatedTuples(testCasesStr)) {
+      TestCases testCase = TestCases.fromString(nameWeight.get(0));
+      int weight = Integer.valueOf(nameWeight.get(1));
+      testCaseWeightPairs.add(new TestCaseWeightPair(testCase, weight));
+    }
+
+    return testCaseWeightPairs;
+  }
+
+  private static List<List<String>> parseCommaSeparatedTuples(String str) {
+    List<List<String>> tuples = new ArrayList<List<String>>();
+    for (String tupleStr : on(',').split(str)) {
```

Comment:
Optional: Stylists consider Joiner.on to be a constant, and just instantiate a single private static instance.   Consider doing something like COMMA_SPLITTER.      It also eases ignoring whitespace, which people typically chain with `.on()`